common/djangoapps/third_party_auth/tests/specs/test_testshib.py

```diff
@@ -175,12 +175,54 @@ def test_registration_form_has_data_sharing_consent_when_optional(self):
         )
         self.assertEqual(ajax_register_response.status_code, 200)
 
+    def test_registration_form_has_data_sharing_consent_when_required_but_not_must(self):
```

Comment:
Method name could use work. Possibly just change it to `...required_but_not_enforced`.